lib/oneview-sdk/resource/storage_system.rb

```diff
@@ -59,13 +60,12 @@ def add
     # @note name,uri or ip_hostname must be specified inside resource
     # @return [Boolean] Whether or not resource exists
     def exists?
-      ip_hostname = self['credentials'][:ip_hostname] || self['credentials']['ip_hostname'] if self['credentials']
-      return true if @data['name'] && self.class.find_by(@client, name: @data['name']).size == 1
-      return true if @data['uri']  && self.class.find_by(@client, uri:  @data['uri']).size == 1
-      return true if ip_hostname && self.class.find_by(@client, credentials: { ip_hostname: ip_hostname }).size == 1
-      unless @data['name'] || @data['uri'] || ip_hostname
-        fail IncompleteResource, 'Must set resource name, uri or ip_hostname before trying to retrieve!'
-      end
+      super
+    rescue IncompleteResource => e
```

Comment:
Oops. This only searches by ip_hostname if the name and uri are not set. It should be able to search by all 3. Same for ServerHardware